amethyst_utils/src/ortho_camera.rs

```diff
@@ -0,0 +1,206 @@
+    Lossy {stretch_direction: Axis2},
+    
+    /// Scales the render dynamically to ensure no space is lost in the [0,1] range on any axis.
+    /// In other words, this ensures that you can always at least see everything 
+    /// between the world coordinates (0, 0) up to (1, 1).
+    /// If you have a non-default `Transform` on your camera,
+    /// it will just translate those coordinates by the translation of the `Transform`.
+    Shrink,
+}
+
+impl CameraNormalizeMode {
+    /// Get the camera matrix offsets according to the specified options.
+    fn camera_offsets(&self, aspect_ratio: f32) -> (f32,f32,f32,f32) {
+        match self {
+            &CameraNormalizeMode::Lossy {ref stretch_direction} => {
+                match stretch_direction {
+                    Axis2::X => {
+                        CameraNormalizeMode::lossy_x(aspect_ratio)
+                    },
+                    Axis2::Y => {
+                        CameraNormalizeMode::lossy_y(aspect_ratio)
+                    },
+                }
+            },
+            &CameraNormalizeMode::Shrink => {
+                if aspect_ratio > 1.0 {
+                    CameraNormalizeMode::lossy_x(aspect_ratio)
+                } else if aspect_ratio < 1.0 {
+                    CameraNormalizeMode::lossy_y(aspect_ratio)
+                } else {
+                    (0.0,1.0,0.0,1.0)
+                }
+            },
+        }
+    }
+    
+    fn lossy_x(aspect_ratio: f32) -> (f32,f32,f32,f32) {
+        let offset = (aspect_ratio - 1.0) / 2.0;
+        (-offset, 1.0 + offset, 0.0, 1.0)
+    }
+
+    fn lossy_y(aspect_ratio: f32) -> (f32,f32,f32,f32) {
+        let offset = (1.0 / aspect_ratio - 1.0) / 2.0;
+        (0.0, 1.0, -offset, 1.0 + offset)
+    }
+}
+
+impl Default for CameraNormalizeMode {
+    fn default() -> Self {
+        CameraNormalizeMode::Shrink
+    }
+}
+
+/// System that automatically changes the camera matrix according to the settings in
+/// the `CameraNormalOrtho` attached to the camera entity.
+#[derive(Default)]
+pub struct CameraNormalOrthoSystem {
+    aspect_ratio_cache: f32,
+}
+
+impl<'a> System<'a> for CameraNormalOrthoSystem {
+    type SystemData = (ReadExpect<'a, ScreenDimensions>, WriteStorage<'a, Camera>, ReadStorage<'a, CameraNormalOrtho>);
+    fn run(&mut self, (dimensions, mut cameras, ortho_cameras): Self::SystemData) {
+        let aspect = dimensions.aspect_ratio();
+        if aspect != self.aspect_ratio_cache {
+            self.aspect_ratio_cache = aspect;
+
+            for (mut camera, ortho_camera) in (&mut cameras, &ortho_cameras).join() {
+                let offsets = ortho_camera.camera_offsets(aspect);
+                camera.proj = Ortho {
+                    left: offsets.0,
+                    right: offsets.1,
+                    bottom: offsets.2,
+                    top: offsets.3,
+                    near: 0.1,
+                    far: 2000.0,
```

Comment:
These values should probably reuse from the previous values in case the user has something custom going on.

Reply:
I'm not smart enough to resolve the formula to get near and far back from the matrix, so I just skipped this one and left a comment :P